File: front-ParkWise/src/app/models/PlaceParking.ts
import { Bloc } from "./Bloc";
import { Etage } from "./etage";
import { Reservation } from "./reservation";

export class PlaceParking {
    id!: number;
    numeroPlace!: string;
    estAccessibleHandicap!: boolean;
    estReservee!: boolean;
    typePlace!: string;
    bloc!: Bloc;
    reservation!:Reservation[];
    isSelected: boolean=false;
    x!: number;  
    y!: number; 
    isDisabled?:boolean;


}

File: backoffice/src/main/java/com/example/achref/Entities/parking/PlaceParking.java
package com.example.achref.Entities.parking;
import com.example.achref.Entities.reservation.Reservation;

import java.io.Serializable;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;


@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Entity
public class PlaceParking implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String numeroPlace;
    private boolean estAccessibleHandicap;
    private boolean estReservee;
    private String typePlace;//pour les cadre, les employees etc...

    @ManyToOne
    @JoinColumn(name = "bloc_id")
    private Bloc bloc;

    @OneToMany(mappedBy = "placeParking", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnore
    private List<Reservation> reservations;
}